### PR TITLE
Service Points feature overview

### PR DESCRIPTION
**Pull request by andriispryker**
Updated documentation to include recommended customizations in order to use the feature in production.

## PR Description

TBD

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md [hunk 1]
```diff
@@ -64,6 +65,35 @@ After placing an order, the customer can see the selected service point on the O
 
 ![Storefront order with a service point](https://spryker.s3.eu-central-1.amazonaws.com/docs/pbc/all/service-point-management/unified-commerce/service-points-feature-overview.md/storefront-order-service-point.png)
 
+
+## Current constraints
+
+* Currently, only product offers are supported, which means that services can be configured exclusively at the level of a product offer
+* Product Catalog is not supported, meaning it can not be filtered by the service type or service provided in the specific service point
+* Product details page is not supported, meaning it will show all product offers created in the system for the product
```
Comment (andriitserkovnyi) [inline docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md:73]: How's this a constraint in the context of service points?

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md [hunk 2]
```diff
@@ -64,6 +65,35 @@ After placing an order, the customer can see the selected service point on the O
...
+* Product can not be added to the shopping cart with pre-selected service point, this can be done only during checkout process
+* If a product was added to the shopping cart without a product offer attached to it then this product can not be purchased with a shipment type other than *Delivery*.
+
+
+Similar  constraints apply to the *shipment type*. See [Shipment feature overview](/docs/pbc/all/carrier-management/{{page.version}}/base-shop/shipment-feature-overview.html).
```
Comment (andriitserkovnyi) [inline docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md:78]: From my pov, it makes more sense to add constraints to the Shipment feature overview than having this message here.

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md [hunk 3]
```diff
@@ -64,6 +65,35 @@ After placing an order, the customer can see the selected service point on the O
...
+
+## Recommended customizations for Click & Collect setup
+
+Since we do not yet support a product catalog or a product details page, the following project-level customizations are recommended on the Product Details Page:
+
+* First, you need to implement the Default Product Offer Reference Strategy if you want a specific offer to be prioritized at the first position. Refer to the [Install the Marketplace Product Offer features](https://docs.spryker.com/docs/pbc/all/offer-management/202307.0/marketplace/install-and-upgrade/install-features/install-the-marketplace-product-offer-feature.html#set-up-behavior)  development documentation for more details. Ensure that only product offers are displayed on prodcut details page.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md:84]: If I don't want a specific offer to be prioritized, do I need to do this?

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md [hunk 4]
```diff
@@ -64,6 +65,35 @@ After placing an order, the customer can see the selected service point on the O
...
+
+* First, you need to implement the Default Product Offer Reference Strategy if you want a specific offer to be prioritized at the first position. Refer to the [Install the Marketplace Product Offer features](https://docs.spryker.com/docs/pbc/all/offer-management/202307.0/marketplace/install-and-upgrade/install-features/install-the-marketplace-product-offer-feature.html#set-up-behavior)  development documentation for more details. Ensure that only product offers are displayed on prodcut details page. 
+* Second, you need to filter the list of product offers displayed on the Product Details page by overriding the Merchant Product Offer Widget behavior. This filter should display product offers with an empty service and shipment type set to _empty_ or _Delivery_. In this case, customers will always be able to buy a product with the default shipment option and, if desired, choose Pickup during checkout." Refer to the [](https://docs.spryker.com/docs/pbc/all/offer-management/202307.0/marketplace/render-merchant-product-offers-on-the-storefront.html#prerequisites).
```
Comment (andriitserkovnyi) [inline docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md:85]: What information are we referring to in the linked doc?

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md [hunk 5]
```diff
@@ -64,6 +65,35 @@ After placing an order, the customer can see the selected service point on the O
...
+* Currently, only product offers are supported, which means that services can be configured exclusively at the level of a product offer
+* Product Catalog is not supported, meaning it can not be filtered by the service type or service provided in the specific service point
+* Product details page is not supported, meaning it will show all product offers created in the system for the product
+* Product can not be added to the shopping cart with pre-selected service point, this can be done only during checkout process
+* If a product was added to the shopping cart without a product offer attached to it then this product can not be purchased with a shipment type other than *Delivery*.
+
+
+Similar  constraints apply to the *shipment type*. See [Shipment feature overview](/docs/pbc/all/carrier-management/{{page.version}}/base-shop/shipment-feature-overview.html).
+
+## Recommended customizations for Click & Collect setup
```
Comment (andriitserkovnyi) [inline docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md:80]: Do you think this info would make more sense in the Enable Click&Collect doc?

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md [hunk 6]
```diff
@@ -64,6 +65,35 @@ After placing an order, the customer can see the selected service point on the O
...
+* Second, you need to filter the list of product offers displayed on the Product Details page by overriding the Merchant Product Offer Widget behavior. This filter should display product offers with an empty service and shipment type set to _empty_ or _Delivery_. In this case, customers will always be able to buy a product with the default shipment option and, if desired, choose Pickup during checkout." Refer to the [](https://docs.spryker.com/docs/pbc/all/offer-management/202307.0/marketplace/render-merchant-product-offers-on-the-storefront.html#prerequisites).
+* Add information about product availability in the service point locations by using Service Point Widget. Refer to [Install the Service Points feature](https://docs.spryker.com/docs/pbc/all/service-point-management/202311.0/unified-commerce/install-features/install-the-service-points-feature.html) development documentation for more details.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md:86]: In the Install the Service Points feature, the installation of the widget is not optional. So why are we referring to this as a customization when it comes as part of the feautre?

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-cart-checkout-feature-overview.md [hunk 7]
```diff
@@ -6,10 +6,34 @@ template: concept-topic-template
 
 This feature provides an example of a replacement strategy used by Click & Collect. The strategy is needed to replace a product offer with another product offer after a customer changes a service point or a shipment type in cart.
 
+How does it work?
+* When a customer changes a shipment type and/or a service point, the replacement strategy is triggered.
+* The strategy applies the selected shipment type and/or service point to the product offers in the system. These are available for the selected product SKU and Merchant linked to the product offer.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-cart-checkout-feature-overview.md:11]: Please clarify what you mean by the second sentence here.

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-cart-checkout-feature-overview.md [hunk 8]
```diff
@@ -6,10 +6,34 @@ template: concept-topic-template
...
+
+
 On the project level, you can extend this feature to support more complex scenarios, like the following:
 * Getting information from external systems about the offer you want to replace the current offer with.
 * Have different algorithms for replacement based on information obtained from a customer.
 
+## Current constraints
+
+* The example strategy does not support a use case where the same product SKU from the same merchant is added to the shopping cart as two separate cart line items. We recommend implementing the following logic on a project level if you intend to use the example strategy in production:
+* * When a product is added to the shopping cart, if the same product from the same merchant is already present, the quantity will be updated automatically, regardless of the specific product offer.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-cart-checkout-feature-overview.md:34]: Is this point part of the previous one?

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-cart-checkout-feature-overview.md (hunk 7 above)
Comment (andriispryker) [reply docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-cart-checkout-feature-overview.md:11]: The SKU and Merchant of the current product offer will be applied as a filter to search the target offer.

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-cart-checkout-feature-overview.md (hunk 8 above)
Comment (andriispryker) [reply docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-cart-checkout-feature-overview.md:34]: Yes, this is a description of what needs to be done. Please add reference to this one here too: https://github.com/spryker/spryker-docs/pull/2266#pullrequestreview-1786418310

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md (hunk 1 above)
Comment (andriispryker) [reply docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md:73]: We may call it feature gap, basically these are the things that are not yet requested/implemented.

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md (hunk 2 above)
Comment (andriispryker) [reply docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md:78]: Ok.

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md (hunk 3 above)
Comment (andriispryker) [reply docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md:84]: Last sentence also requires cusotmization, i would say then separate them.

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md (hunk 4 above)
Comment (andriispryker) [reply docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md:85]: All of it, developer just need to figure it out based on the doc that explains how widget works, we did not develop a step by step guide nor sample code for this one.

File: docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md (hunk 6 above)
Comment (andriispryker) [reply docs/pbc/all/service-point-management/202311.0/unified-commerce/service-points-feature-overview.md:86]: The widget OOTB displayed only on checkout, but on PDP you need to add some code to call for it, this is just a suggested / recommended way to build an end-to-end UX. We could not do it OOTB cos we keep vanilla code base clean from specific implementations. Customer can choose to go with more sofisticated approuch if they want to. We just saying this is a easiest one with almost no effrots, other UX require more changes on the project level.

**Review by andriispryker — COMMENTED** [review]
Done.

**Pull request merged**